core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -407,6 +407,10 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope)
             }
             accessControl.checkCanInsertIntoTable(session.toSecurityContext(), targetTable);
 
+            if (!accessControl.getRowFilters(session.toSecurityContext(), targetTable).isEmpty()) {
+                throw semanticException(NOT_SUPPORTED, insert, "Insert into table with a row filter is not supported");
```

Comment:
or "is not allowed"

Reply:
https://github.com/trinodb/trino/issues/7369